rotkehlchen/tests/unit/test_tokens.py

```diff
@@ -140,3 +145,50 @@ def test_last_queried_ts(tokens, freezer):
         assert len(after_second_query) == 1
         assert after_second_query[0][0] == 'last_queried_timestamp'
         assert int(after_second_query[0][1]) >= continuation
+
+
+def test_cache_is_per_token_type(ethereum_inquirer):
+    """This test makes sure that different info cache is used per token type."""
+    address = make_evm_address()
+
+    def query_token_info(token_kind):
+        """
+        Util function to request token info. Doesn't pass name, symbol or decimals because they
+        should be retrieved from the chain (chain calls are mocked below).
+        """
+        return _query_or_get_given_token_info(
+            evm_inquirer=ethereum_inquirer,
+            evm_address=address,
+            name=None,
+            symbol=None,
+            decimals=None,
+            token_kind=token_kind,
+        )
+
+    def patch_multicall_2(return_value):
+        """
+        This patch method together with ERC20_INFO_RESPONSE and ERC721_INFO_RESPONSE mocks
+        tokens info.
+        """
+        return patch.object(
+            ethereum_inquirer,
+            'multicall_2',
+            return_value=return_value,
+        )
+
+    with patch_multicall_2(ERC20_INFO_RESPONSE):
+        erc20_token_data = query_token_info(EvmTokenKind.ERC20)
+
+    with patch_multicall_2(ERC721_INFO_RESPONSE):
+        erc721_token_data = query_token_info(EvmTokenKind.ERC721)
+
+    with patch.object(  # disable chain calls
```

Comment:
comment elaboration as it took me some time to understand what you do here

```suggestion
    with patch.object(  # fail if cache is not used and chain is called
```